packages/flutter/lib/src/material/data_table.dart

```diff
@@ -393,7 +393,13 @@ class DataTable extends StatelessWidget {
     this.onSelectAll,
     this.decoration,
     this.dataRowColor,
+    @Deprecated(
+      'Migrate to use dataRowMinHeight and dataRowMaxHeight instead. '
+      'This feature was deprecated after v3.5.0-10.0.pre.',
+    )
     this.dataRowHeight,
+    this.dataRowMinHeight,
```

Comment:
Below there should be some asserts that ensure both options (the old and new) aren't used together.